### PR TITLE
Support MinGW

### PR DESCRIPTION
**Pull request by orgads**

### CODE REVIEW

File: subprocess.h [hunk 1]
```diff
@@ -44,6 +44,11 @@
 #include <stdio.h>
 #include <string.h>
 
+#ifdef _WIN32
+#include <windows.h>
```
Comment (sheredom) [inline subprocess.h:48]: No-way am I including `windows.h` in this library, it bloats compile time by 10x.

File: subprocess.h [hunk 2]
```diff
@@ -306,63 +313,6 @@ struct subprocess_overlapped_s {
   void *hEvent;
 };
 
-#pragma warning(pop)
```
Comment (sheredom) [inline subprocess.h:309]: This was all done for a very specific reason - namely to make compile time fast, no way I'll accept removing these!

**Review by sheredom — CHANGES_REQUESTED** [review]
Overall this PR is not able to be merged with the `windows.h` inclusion.

File: subprocess.h (hunk 1 above)
Comment (orgads) [reply subprocess.h:48]: I thought so.

I'll try to find a fix then. Getting error 87 on CreateProcess, which means the function signature is wrong.

**Pull request closed without merging**